src/journal/journal-verify.c

```diff
@@ -744,6 +744,14 @@ static int verify_entry_array(
                         draw_progress(0x8000 + scale_progress(0x3FFF, i, n), last_usec);
 
                 if (a == 0) {
+                        /* One unreferenced entry object is OK since it can happen if journald were
+                         * about to append a new entry but there was only enough space to allocate
+                         * the entry object itself, i.e. no more room for the items nor a new entry
+                         * array. */
+                        if (n - i == 1) {
+                                debug(a, "One entry object not referenced in arrays");
+                                break;
+                        }
```

Comment:
I am not sure this is the right "fix". It just tapes over problems.

If we hit this we know that something is off either way, i.e. the journal not fully written beyond this point. Which is entirely OK, but this should be handled systematically, not specifically. i.e. instead of taping over this specific issue, we should propagate out where we exactly found this, and then assume that everything beyond this point is not synced to disk, and we cannot really say much about it.

The current verifier is not smart enough for that, it assumes everything always has hit the disk when things are verified. It shouldn't do that. instead it should be smart enough to determine how far things are entirely clean, and then stop validation, and not complain the rest was borked, but just that it can't come up with a clear opinion on the rest, because partially synced only.

Reply:
yeah... that's a pretty hackish and limited workaround for a general shortcoming of the journal verifier. I think it's worth to open a bug report to discuss this specific issue.

Reply:
@poettering please a look at PR #18057 for a different possible approach.